### PR TITLE
Fixes #714 assertTrue -> assertEquals

### LINKED ISSUE #714: robot-simulator: use assertEquals instead of assertTrue
In RobotTests, `assertTrue(expected.equals(actual))` is used for all assertions. This should be changed to `assertEquals(expected, actual)` because it is more readable and gives better error messages when the tests fail.

### PR DESCRIPTION
**Pull request by matthewstyler**
---

Reviewer Resources:

[Track Policies](https://github.com/exercism/xjava/blob/master/POLICIES.md#event-checklist)

### CODE REVIEW

**Comment by matthewstyler** [comment]
tests failing, will investigate

**Comment by stkent** [comment]
I think #714 is invalid; there's a deliberate use of a custom `equals` method there. From the `GridPosition` starter class:

```java
class GridPosition {

    final int x;

    final int y;

    GridPosition(final int x, final int y) {
        this.x = x;
        this.y = y;
    }

    /*
     * This equals method is of deliberately narrow scope (only allows comparison with another GridPosition) to increase
     * readability. In general, one should provide a full implementation of Object.equals(Object obj) and a
     * corresponding implementation of Object.hashCode(). See
     *
     * https://docs.oracle.com/javase/7/docs/api/java/lang/Object.html#equals(java.lang.Object)
     *
     * and
     *
     * https://docs.oracle.com/javase/7/docs/api/java/lang/Object.html#hashCode()
     *
     * for more information.
     */
    boolean equals(final GridPosition gridPosition) {
        return this.x == gridPosition.x && this.y == gridPosition.y;
    }

}
```

**Comment by stkent** [comment]
We could instead provide a full equals implementation, just need to be wary of the fact that it'll be visible to implementors.

**Comment by Smarticles101** [comment]
So then is the assertEquals method using the generic `equals` method from `java.lang.Object`?

**Comment by stkent** [comment]
Yep, exactly, hence the failures!

**Comment by matthewstyler** [comment]
My vote would be for an overridden equals, the class only exists to report the position

**Comment by FridaTveit** [comment]
Sorry, that's my bad for not checking throroughly enough when creating the issue! I did check that the change worked locally, but only for some of the tests when the grid position didn't actually change. 

I agree, I think overriding equals would be the best plan :) Would you be happy doing that as part of this PR @matthewstyler? You absolutely don't have to, I'm happy to open a separate PR for it :)

**Comment by matthewstyler** [comment]
ya, no problem @FridaTveit

File: exercises/robot-simulator/src/example/java/GridPosition.java [hunk 1]
```diff
@@ -9,21 +9,26 @@
...
-     * and
-     *
-     * https://docs.oracle.com/javase/7/docs/api/java/lang/Object.html#hashCode()
-     *
-     * for more information.
-     */
-    boolean equals(final GridPosition gridPosition) {
-        return this.x == gridPosition.x && this.y == gridPosition.y;
-    }
+    @Override
```
Comment (FridaTveit) [inline exercises/robot-simulator/src/example/java/GridPosition.java]: The spacing seems to have gone a bit wrong here and in the test file. It should be indented with four spaces :) If you're using IntelliJ you can use `ctrl + alt + L` to automatically format the code (`cmd + alt + L` on a mac). That should fix spacing issues in the current file and will even remove unused imports if you tell it to in the reformat code dialog box (see https://www.jetbrains.com/help/idea/reformatting-source-code.html). On eclipse it's `ctrl + shift + F` :)

**Comment by Smarticles101** [comment]
Thanks for taking the time to do all this @matthewstyler I think it looks good now

**Pull request merged**